src/main/java/com/thealgorithms/datastructures/graphs/BoruvkaAlgorithm.java

```diff
@@ -0,0 +1,217 @@
+         *
+         * @return an array containing the cheapest edge for each subset.
+         */
+        private Edge[] computeCheapestEdges() {
+            Edge[] cheapest = new Edge[graph.vertex];
+            for (final var edge : graph.edges) {
+                final var set1 = find(components, edge.src);
+                final var set2 = find(components, edge.dest);
+
+                if (set1 != set2) {
+                    if (cheapest[set1] == null || edge.weight < cheapest[set1].weight) {
+                        cheapest[set1] = edge;
+                    }
+                    if (cheapest[set2] == null || edge.weight < cheapest[set2].weight) {
+                        cheapest[set2] = edge;
+                    }
+                }
+            }
+            return cheapest;
+        }
+    }
+
+    /**
+     * Finds the parent of the subset using path compression
+     *
+     * @param components array of subsets
+     * @param i          index of the subset
+     * @return the parent of the subset
+     */
+    static int find(final Component[] components, final int i) {
+        if (components[i].parent != i) {
+            components[i].parent = find(components, components[i].parent);
+        }
+        return components[i].parent;
+    }
+
+    /**
+     * Performs the Union operation for Union-Find
+     *
+     * @param components array of subsets
+     * @param x          index of the first subset
+     * @param y          index of the second subset
+     */
+    static void union(Component[] components, final int x, final int y) {
+        final int xroot = find(components, x);
+        final int yroot = find(components, y);
+
+        if (components[xroot].rank < components[yroot].rank) {
+            components[xroot].parent = yroot;
+        } else if (components[xroot].rank > components[yroot].rank) {
+            components[yroot].parent = xroot;
+        } else {
+            components[yroot].parent = xroot;
+            components[xroot].rank++;
+        }
+    }
+
+    /**
+     * Boruvka's algorithm to find the Minimum Spanning Tree
+     *
+     * @param graph the graph
+     * @return list of edges in the Minimum Spanning Tree
+     */
+    static List<Edge> boruvkaMST(final Graph graph) {
+        var boruvkaState = new BoruvkaState(graph);
+
+        while (boruvkaState.hasMoreEdgesToAdd()) {
+            final var cheapest = boruvkaState.computeCheapestEdges();
+            boruvkaState.merge(cheapest);
+        }
+        return boruvkaState.result;
+    }
+
+    /**
+     * Initializes subsets for Union-Find
+     *
+     * @param graph the graph
+     * @return the initialized subsets
+     */
+    private static Component[] initializeSubsets(final Graph graph) {
+        Component[] components = new Component[graph.vertex];
+        for (int v = 0; v < graph.vertex; ++v) {
+            components[v] = new Component(v, 0);
+        }
+        return components;
+    }
```

Comment:
This should be a member of `BoruvkaState`. Also the name could be changed to `initializeComponents` or `getNodesAsComponents`.